components/table/table-wrapper.js

```diff
@@ -0,0 +1,318 @@
+	}
+	d2l-table-wrapper[type="default"][dir="rtl"] .d2l-table > * > tr > *,
+	d2l-table-wrapper[type="default"]:not([dir="rtl"]) .d2l-table > * > tr > .d2l-table-cell-first {
+		border-left: var(--d2l-table-border);
+	}
+	d2l-table-wrapper[type="default"] .d2l-table-row-first > * {
+		border-top: var(--d2l-table-border); /* add top border to default first row */
+	}
+
+	/* header cells */
+	.d2l-table > thead > tr > th,
+	.d2l-table > * > tr[header] > th {
+		background-color: var(--d2l-table-header-background-color);
+		font-size: 0.7rem;
+		line-height: 0.9rem;
+	}
+	d2l-table-wrapper[type="default"] .d2l-table > thead > tr > th,
+	d2l-table-wrapper[type="default"] .d2l-table > * > tr[header] > th {
+		height: 27px; /* min-height to be 48px including border */
+	}
+
+	/* border radiuses */
+	d2l-table-wrapper[type="default"]:not([dir="rtl"]) .d2l-table-row-first > .d2l-table-cell-first,
+	d2l-table-wrapper[type="default"][dir="rtl"] .d2l-table-row-first > .d2l-table-cell-last {
+		border-top-left-radius: var(--d2l-table-border-radius);
+	}
+	d2l-table-wrapper[type="default"]:not([dir="rtl"]) .d2l-table-row-first > .d2l-table-cell-last,
+	d2l-table-wrapper[type="default"][dir="rtl"] .d2l-table-row-first > .d2l-table-cell-first {
+		border-top-right-radius: var(--d2l-table-border-radius);
+	}
+	d2l-table-wrapper[type="default"]:not([dir="rtl"]) .d2l-table-row-last > .d2l-table-cell-first,
+	d2l-table-wrapper[type="default"][dir="rtl"] .d2l-table-row-last > .d2l-table-cell-last {
+		border-bottom-left-radius: var(--d2l-table-border-radius);
+	}
+	d2l-table-wrapper[type="default"]:not([dir="rtl"]) .d2l-table-row-last > .d2l-table-cell-last,
+	d2l-table-wrapper[type="default"][dir="rtl"] .d2l-table-row-last > .d2l-table-cell-first {
+		border-bottom-right-radius: var(--d2l-table-border-radius);
+	}
+
+	/* selected rows */
+	.d2l-table > tbody > tr[selected] {
+		background-color: var(--d2l-table-row-background-color-selected);
+	}
+	d2l-table-wrapper[type="default"]:not([dir="rtl"]) .d2l-table > tbody > tr[selected] > .d2l-table-cell-last,
+	d2l-table-wrapper[type="default"][dir="rtl"] .d2l-table > tbody > tr[selected] > .d2l-table-cell-first {
+		border-right-color: var(--d2l-table-row-border-color-selected);
+	}
+	d2l-table-wrapper[type="default"]:not([dir="rtl"]) .d2l-table > tbody > tr[selected] > .d2l-table-cell-first,
+	d2l-table-wrapper[type="default"][dir="rtl"] .d2l-table > tbody > tr[selected] > .d2l-table-cell-last {
+		border-left-color: var(--d2l-table-row-border-color-selected);
+	}
+	.d2l-table > tbody > tr[selected] > *,
+	.d2l-table > * > tr.d2l-table-selected-previous > * {
+		border-bottom-color: var(--d2l-table-row-border-color-selected);
+	}
+	.d2l-table > * > tr.d2l-table-selected-first > * {
+		border-top: 1px solid var(--d2l-table-row-border-color-selected);
+	}
+
+	/* no-column-border */
+	d2l-table-wrapper[type="default"]:not([dir="rtl"]) .d2l-table[no-column-border] > tbody > tr > *:not(.d2l-table-cell-last) {
```

Comment:
This `no-column-border` thing is something I want to move to `<d2l-table-wrapper>` -- will investigate as a future change.